lib/Db/TimelineQueryFilters.php

```diff
@@ -37,6 +42,16 @@ public function transformVideoFilter(IQueryBuilder &$query, bool $aggregate): vo
         $query->andWhere($query->expr()->eq('m.isvideo', $query->expr()->literal(1)));
     }
 
+    public function transformLivePhotoFilter(IQueryBuilder &$query, bool $aggregate): void
+    {
+        $query->andWhere($query->expr()->neq('m.liveid', $query->expr()->literal('')));
+    }
+
+    public function transformPanoFilter(IQueryBuilder &$query, bool $aggregate): void
+    {
+        $query->andWhere('m.w >= '.PANOROMA_ASPECT_RATIO.' * m.h');
```

Comment:
Both of these will scan the entire table - these need the right indexes.

Reply:
AFAIK this expression can't be optimized with index too, probably needs a separate boolean field.

Reply:
thanks for the feedback, makes sense! If I find some time later, I'll try to add a `pano` boolean field in the `oc_memories` table